From 11ed5626c8b2e6c87d5c7f5afc7aad1780189574 Mon Sep 17 00:00:00 2001
From: Ido Reis <idox.reis@intel.com>
Date: Mon, 5 Sep 2016 13:10:14 +0300
Subject: [PATCH] hostapd: probe req dbg cleanup

Signed-off-by: Ido Reis <idox.reis@intel.com>
---
 src/ap/drv_callbacks.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/ap/drv_callbacks.c b/src/ap/drv_callbacks.c
index 12db25f..c9a7457 100644
--- a/src/ap/drv_callbacks.c
+++ b/src/ap/drv_callbacks.c
@@ -1185,7 +1185,9 @@ void wpa_supplicant_event(void *ctx, enum wpa_event_type event,
 		    WLAN_FC_GET_STYPE(fc) == WLAN_FC_STYPE_PROBE_REQ)
 			level = MSG_EXCESSIVE;
 	}
-
+	if (event == EVENT_RX_PROBE_REQ) {
+		level = MSG_MSGDUMP;
+	}
 	wpa_dbg(hapd->msg_ctx, level, "Event %s (%d) received",
 		event_to_string(event), event);
 #endif /* CONFIG_NO_STDOUT_DEBUG */
-- 
2.10.1

