From e8d135acfda87f13a29c0398d198e531cffd39d5 Mon Sep 17 00:00:00 2001
From: Simon Dinkin <simonx.dinkin@intel.com>
Date: Sun, 7 Aug 2016 12:21:43 +0300
Subject: [PATCH] disable new implemantation of proxy arp

---
 src/ap/hostapd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index a43fa96..297cd3a 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -1134,7 +1134,7 @@ static int hostapd_setup_bss(struct hostapd_data *hapd, int first)
 		wpa_printf(MSG_ERROR, "BSS Load initialization failed");
 		return -1;
 	}
-
+#if 0
 	if (conf->proxy_arp) {
 		if (x_snoop_init(hapd)) {
 			wpa_printf(MSG_ERROR,
@@ -1154,7 +1154,7 @@ static int hostapd_setup_bss(struct hostapd_data *hapd, int first)
 			return -1;
 		}
 	}
-
+#endif /* disable proxy arp */
 	if (!hostapd_drv_none(hapd) && vlan_init(hapd)) {
 		wpa_printf(MSG_ERROR, "VLAN initialization failed.");
 		return -1;
-- 
2.10.1

