From 6ec84c9e1b42103770f6d745e88c342631380fdc Mon Sep 17 00:00:00 2001
From: pukitis <martins.pukitis@intel.com>
Date: Tue, 14 Aug 2018 16:24:30 +0300
Subject: [PATCH] disable_L2_wnm_keepalive_frames

---
 wpa_supplicant/events.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/wpa_supplicant/events.c b/wpa_supplicant/events.c
index fe5ab5b..68bf40a 100644
--- a/wpa_supplicant/events.c
+++ b/wpa_supplicant/events.c
@@ -1826,14 +1826,16 @@ static void wnm_bss_keep_alive(void *eloop_ctx, void *sock_ctx)
 		return;
 
 	if (!wpa_s->no_keep_alive) {
-		wpa_printf(MSG_DEBUG, "WNM: Send keep-alive to AP " MACSTR,
+		wpa_printf(MSG_DEBUG, "WNM: Skip sending keep-alive to AP " MACSTR,
 			   MAC2STR(wpa_s->bssid));
 		/* TODO: could skip this if normal data traffic has been sent */
 		/* TODO: Consider using some more appropriate data frame for
 		 * this */
+#if 0 /* Skip sending till frame format will be defined */
 		if (wpa_s->l2)
 			l2_packet_send(wpa_s->l2, wpa_s->bssid, 0x0800,
 				       (u8 *) "", 0);
+#endif
 	}
 
 #ifdef CONFIG_SME
-- 
2.10.1

