diff -Naur org/configure.in new/configure.in
--- org/configure.in
+++ new/configure.in
@@ -1,7 +1,7 @@
 
 AC_REVISION($Revision: 1.21 $)
 
-AC_INIT(IFX MEI CPE Driver, 1.7.1-pd2, , drv_mei_cpe)
+AC_INIT(IFX MEI CPE Driver, 1.7.1-pd3, , drv_mei_cpe)
 AC_CONFIG_SRCDIR(src/Makefile.am)
 AM_INIT_AUTOMAKE([tar-pax])
 
diff -Naur org/src/drv_mei_cpe_download_vrx.c new/src/drv_mei_cpe_download_vrx.c
--- org/src/drv_mei_cpe_download_vrx.c
+++ new/src/drv_mei_cpe_download_vrx.c
@@ -1955,12 +1955,12 @@ static IFX_int32_t MEI_VRX_FMLT2_BarRegistersUpdate(
 
    MEI_REG_ACCESS_ME_XMEM_BAR_SET(pMeiDrvCntrl, barIdx,
       (MEI_DRV_PCIE_PHY_MEMBASE_GET(&pMeiDev->meiDrvCntrl) +
-      MEI_CGU_OFFSET + MEI_PLL_OMCFG_OFFSET));
+      MEI_CGU_OFFSET));
    MEI_BAR_TYPE_SET(pMeiDev, barIdx, eMEI_BAR_TYPE_SPECIAL);
    PRN_DBG_USR_NL( MEI_DRV, MEI_DRV_PRN_LEVEL_NORMAL,
       ("BAR[%02d] = 0x%08x (-> CGU / PLL_OMCFG)" MEI_DRV_CRLF, barIdx,
       (MEI_DRV_PCIE_PHY_MEMBASE_GET(&pMeiDev->meiDrvCntrl) +
-      MEI_CGU_OFFSET + MEI_PLL_OMCFG_OFFSET)));
+      MEI_CGU_OFFSET)));
 
    pMeiDev->barSafeAddr = pChunk[last_used_chunkIdx].pBARx;
 
