diff -Naur org/src/drv_mei_cpe_device_vrx.c new/src/drv_mei_cpe_device_vrx.c
--- org/src/drv_mei_cpe_device_vrx.c	2014-09-15 23:21:30.000000000 +0200
+++ new/src/drv_mei_cpe_device_vrx.c	2015-01-21 03:04:39.861143000 +0100
@@ -135,8 +135,8 @@
 
    pMeiDrvCntrl->MEI_pcie_phy_membase = (IFX_uint32_t)MEI_pcie_ep_dev.phy_membase;
    pMeiDrvCntrl->MEI_pcie_virt_membase = (IFX_uint32_t)MEI_pcie_ep_dev.membase;
-   pMeiDrvCntrl->MEI_pcie_irq = MEI_pcie_ep_dev.irq;
-
+   /* Use active polling mode */
+   pMeiDrvCntrl->MEI_pcie_irq = 99;
    return IFX_SUCCESS;
 }
 
