--- org/configure.in
+++ new/configure.in
@@ -1,7 +1,7 @@
 
 AC_REVISION($Revision: 1.21 $)
 
-AC_INIT(IFX MEI CPE Driver, 1.7.2-pd1, , drv_mei_cpe)
+AC_INIT(IFX MEI CPE Driver, 1.7.2-pd2, , drv_mei_cpe)
 AC_CONFIG_SRCDIR(src/Makefile.am)
 AM_INIT_AUTOMAKE([tar-pax])
 
--- org/src/drv_mei_cpe_msg_process.c
+++ new/src/drv_mei_cpe_msg_process.c
@@ -1524,6 +1524,7 @@ MEI_STATIC IFX_void_t MEI_RecvAutonomMai
 
    MEI_IF_STAT_INC_RECV_NFC_COUNT(pMeiDev);
    MEI_DBG_MSG_LOG_DUMP(pMeiDev, &tempMsg.mbCmv.cmv);
+   MEI_LOG_CMV_MSG( pMeiDev, (CMV_STD_MESSAGE_T *)&tempMsg, "OTHER CMV", MEI_DRV_PRN_LEVEL_NORMAL);
 
    /* check for special notifications */
    switch (CMV_MSGHDR_MSGID_GET(tempMsg.mbCmv.cmv))
